extensions/notebook/src/common/utils.ts

```diff
@@ -325,16 +325,47 @@ export async function getRandomToken(size: number = 24): Promise<string> {
 }
 
 export function isBookItemPinned(notebookPath: string): boolean {
-	let pinnedNotebooks: string[] = getPinnedNotebooks();
-	if (pinnedNotebooks?.indexOf(notebookPath) > -1) {
+	let pinnedNotebooks: IBookNotebook[] = getPinnedNotebooks();
+	if (pinnedNotebooks?.find(x => x.notebookPath === notebookPath)) {
 		return true;
 	}
 	return false;
 }
 
-export function getPinnedNotebooks(): string[] {
+export function getPinnedNotebooks(): IBookNotebook[] {
 	let config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(notebookConfigKey);
-	let pinnedNotebooks: string[] = config.get(pinnedBooksConfigKey) ?? [];
+	let pinnedNotebooks: [] = config.get(pinnedBooksConfigKey);
+	let pinnedBookDirectories: IBookNotebook[] = [];
+	let updateFormat: boolean = false;
+	pinnedNotebooks.map((elem, index) => {
```

Comment:
Either use foreach or something like
```
const pinnedBookDirectories = pinnedNotebooks.map(elem => {
if(typeof(elem) === 'string') {
   updateFormat = true;
   return { notebookPath: elem, bookPath: '' };
} else {
   return elem as IBookNotebook;
});
```

Map should only be used if you're actually using the returned array - if not then forEach is better.